src/ray/core_worker/core_worker.cc

```diff
@@ -1556,6 +1556,14 @@ Status CoreWorker::Get(const std::vector<ObjectID> &ids,
   for (const ObjectID &id : ids) {
     if (experimental_mutable_object_provider_->ReaderChannelRegistered(id)) {
       is_experimental_channel = true;
+      continue;
+    }
+
+    Status error_set = experimental_mutable_object_provider_->IsErrorSet(id);
+    if (error_set.IsChannelError()) {
```

Comment:
I feel this experimental_mutable_object_provider_ API is a bit confusing to use.

experimental_mutable_object_provider_ can probably expose a more user-friendly API, something like `bool isChannelClosed()`, and internally you can implement it with `CheckHasError()`